dffml/model/model.py

```diff
@@ -84,6 +106,208 @@ def __call__(self) -> ModelContext:
         self._make_config_location()
         return self.CONTEXT(self)
 
+    async def __aenter__(self):
+        if self.config.location.is_file():
+            temp_dir = self._get_directory()
+            await self._run_operation(
+                "extract", self.config.location, temp_dir
+            )
+            self.location = temp_dir
+
+    async def __aexit__(self, exc_type, exc_value, traceback):
+        if self.config.location.is_file():
+            config_path = self.location / "config.json"
+            config_path.write_text(json.dumps(export(self.config)))
+            await self._run_operation(
+                "archive", self.temp_dir, self.config.location
+            )
```

Comment:
Let's make sure that if this is the first time we are using a model, then we create an archive of the saved state

Reply:
Mentioned related flow [here](https://github.com/intel/dffml/pull/1174#issuecomment-893176136)

Reply:
The flow was implemented and pushed, which handles the first time use case as well using mimetypes...